crates/transaction-pool/src/validate/eth.rs

```diff
@@ -60,27 +60,42 @@ where
     Client: StateProviderFactory,
     Tx: EthPoolTransaction,
 {
-    /// Validates a single transaction.
-    ///
-    /// See also [`TransactionValidator::validate_transaction`]
-    pub fn validate_one(
-        &self,
-        origin: TransactionOrigin,
-        transaction: Tx,
-    ) -> TransactionValidationOutcome<Tx> {
-        self.inner.validate_one(origin, transaction)
-    }
```

Comment:
we still want to keep this but we want a new function that takes a helper where the provider initialization is optional, I assume an `&mut Option<Client>` could work